src/components/BillingDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import { IS_NIGHTLY_OR_DEBUG } from '@src/routes/utils'
+import { useSelector } from '@xstate/react'
+import { openExternalBrowserIfDesktop } from '@src/lib/openWindow'
+import { CustomIcon } from '@src/components/CustomIcon'
+import {
+  BillingRemaining,
+  BillingRemainingMode,
+} from '@src/components/BillingRemaining'
+
+import type { BillingActor } from '@src/machines/billingMachine'
+
+export const BillingDialog = (props: { billingActor: BillingActor }) => {
+  const billingContext = useSelector(
+    props.billingActor,
+    ({ context }) => context
+  )
+  const hasUnlimited = billingContext.credits === Infinity
+
+  if (!IS_NIGHTLY_OR_DEBUG) {
+    return <></>
+  }
+
+  return (
+    <div className="bg-ml-green fg-ml-black flex flex-row rounded-lg p-2 gap-2">
+      <div>
+        <div className="rounded bg-ml-black p-1">
+          {hasUnlimited ? (
+            <CustomIcon name="infinity" />
+          ) : (
+            <CustomIcon name="star" />
+          )}
+        </div>
+      </div>
+      <div className="flex flex-col gap-2">
+        <div className="font-bold font-lg text-ml-black">
```

Comment:
Just some tweaks to spacing and text sizing

```suggestion
    <div className="bg-ml-green fg-ml-black flex flex-row rounded-lg p-4 gap-2 text-xs">
      <div>
        <div className="rounded bg-ml-black p-1">
          {hasUnlimited ? (
            <CustomIcon name="infinity" />
          ) : (
            <CustomIcon name="star" />
          )}
        </div>
      </div>
      <div className="flex flex-col gap-2">
        <div className="font-bold text-ml-black">
```